src/test/validation_chainstatemanager_tests.cpp

```diff
@@ -328,6 +354,31 @@ struct SnapshotTestSetup : TestChain100Setup {
             loaded_snapshot_blockhash);
         return std::make_tuple(&validation_chainstate, &snapshot_chainstate);
     }
+
+    // Simulate a restart of the node by flushing all state to disk, clearing the
+    // existing ChainstateManager, and unloading the block index.
+    //
+    // @returns a reference to the "restarted" ChainstateManager
+    ChainstateManager& SimulateNodeRestart()
+    {
+        ChainstateManager& chainman = *Assert(m_node.chainman);
+
+        BOOST_TEST_MESSAGE("Simulating node restart");
+        {
+            LOCK(::cs_main);
+            for (CChainState* cs : chainman.GetAll()) {
+                cs->ForceFlushStateToDisk();
+            }
+            chainman.ResetChainstates();
+            BOOST_CHECK_EQUAL(chainman.GetAll().size(), 0);
+            const ChainstateManager::Options chainman_opts{
+                ::Params(),
+                static_cast<int64_t(*)()>(::GetTime),
+            };
+            m_node.chainman.reset(new ChainstateManager(chainman_opts));
```

Comment:
In commit "test: add testcases for snapshot initialization" (15da925f2951eaadf81aea9d5b67ec1646581d30)

Might be a little more robust to destroy the old chainman before creating a new one. I.e. add another blank `chainman.reset()` call before this.

Reply:
Fixed, thanks.